pyk/src/pyk/ktool/kprint.py

```diff
@@ -87,11 +88,32 @@ def __init__(self, definition_dir: Path, use_directory: Optional[Path] = None, p
             td = TemporaryDirectory()
             self.use_directory = Path(td.name)
         check_dir_path(self.use_directory)
-        self.definition = readKastTerm(self.definition_dir / 'compiled.json')
-        self.symbol_table = build_symbol_table(self.definition, opinionated=True)
-        self.definition_hash = hash_str(self.definition)
+        self._definition = None
+        self._definition_hash = None
+        self._symbol_table = None
         self._profile = profile
 
+    @property
+    def definition(self) -> KDefinition:
+        if not self._definition:
+            self._definition = readKastTerm(self.definition_dir / 'compiled.json')
+            assert self._definition is not None
+        return self._definition
+
+    @property
+    def definition_hash(self) -> str:
+        if not self._definition_hash:
+            self._definition_hash = hash_str(self.definition)
+            assert self._definition_hash is not None
+        return self._definition_hash
```

Comment:
Did you consider  `self.definition.hash` instead?

Reply:
Good idea, I'll see if it works.

Reply:
Done.